Kernel/Arch/PageFault.h

```diff
@@ -28,8 +29,17 @@ struct PageFaultFlags {
 class PageFault {
 public:
     PageFault(u16 code, VirtualAddress vaddr)
-        : m_code(code)
-        , m_vaddr(vaddr)
+        : m_vaddr(vaddr)
+    {
+        m_type = (Type)(code & 1);
+        m_access = (Access)(code & 2);
+        m_execution_mode = (code & 4) == PageFaultFlags::UserMode ? ExecutionMode::User : ExecutionMode::Kernel;
```

Comment:
I realize this matches the old code, but isn't the usual pattern for testing bit flags `(code & PageFaultFlags::UserMode) != 0`? (Also below.)

Reply:
I'm not sure what the usual pattern is, but changed it to your suggested pattern anyway!